certification/pyxis/pyxis.go

```diff
@@ -0,0 +1,179 @@
+package pyxis
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+const (
+	apiVersion = "v1"
+)
+
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+type pyxisEngine struct {
+	ApiToken  string
+	ProjectId string
+	Client    HTTPClient
+}
+
+func getPyxisUrl(path string) string {
+	return fmt.Sprintf("https://%s/%s/%s", viper.GetString("pyxis-host"), apiVersion, path)
+}
+
+func NewPyxisEngine(apiToken string, projectId string, httpClient HTTPClient) *pyxisEngine {
+	return &pyxisEngine{
+		ApiToken:  apiToken,
+		ProjectId: projectId,
+		Client:    httpClient,
+	}
+}
+
+func (p *pyxisEngine) GetRedHatLayers() (*[]Layer, error) {
```

Comment:
Is this method being used?

Reply:
Not yet. I'll move it to its own file and leave it out of this patch.